WeakAuras/RegionTypes/Icon.lua

```diff
@@ -557,6 +566,11 @@ local function modify(parent, region, data)
   function region:SetInverse(inverse)
     cooldown:SetReverse(not inverse);
   end
+  
+  function region:SetZoom(zoom)
+    data.zoom = zoom;
```

Comment:
That modifies the setting in the actual aura settings.

What needs to happen is to introduce region.zoom, e.g. look at how keepAspectRatio is copied to the region and the Set method only sets that.

You probably should split up UpdateSize() into the parts that modify the widht/height and those that update the tex coordinates, so that you can call the later without the former code.

Reply:
Thanks for the quick response! I will look into it and update it as soon as possible.